crates/libcontainer/src/process/container_main_process.rs

```diff
@@ -183,19 +185,36 @@ pub fn container_main_process(container_args: &ContainerArgs) -> Result<(Pid, bo
 
         #[cfg(feature = "libseccomp")]
         if let Some(seccomp) = linux.seccomp() {
-            let state = crate::container::ContainerProcessState {
-                oci_version: container_args.spec.version().to_string(),
-                // runc hardcode the `seccompFd` name for fds.
-                fds: vec![String::from("seccompFd")],
-                pid: init_pid.as_raw(),
-                metadata: seccomp.listener_metadata().to_owned().unwrap_or_default(),
-                state: container_args
-                    .container
-                    .as_ref()
-                    .ok_or(ProcessError::ContainerStateRequired)?
-                    .state
-                    .clone(),
+            let container = container_args
+                .container
+                .as_ref()
+                .ok_or(ProcessError::ContainerStateRequired)?;
+
+            // Determine OCI status based on container type (matching runc behavior)
+            let oci_status = match container_args.container_type {
+                ContainerType::InitContainer => oci_spec::runtime::ContainerState::Creating,
+                ContainerType::TenantContainer { .. } => oci_spec::runtime::ContainerState::Running,
             };
+
+            // Build OCI-compliant ContainerProcessState using builder pattern
+            let oci_state = oci_spec::runtime::StateBuilder::default()
+                .version(container.state.oci_version.clone())
```

Comment:
In runc, `specs.Version` refers to the Version constant from the github.com/opencontainers/runtime-spec/specs-go package.

specs(specs-go) package: https://github.com/opencontainers/runtime-spec/blob/main/specs-go/version.go

In youki, I think it would correspond to the following Version.

https://github.com/youki-dev/oci-spec-rs/blob/main/src/runtime/version.rs

i.e, `oci_spec::runtime::VERSION`

Reply:
This is a bit outside the scope of this PR, but

`container.state.oci_version` differs from runc.

In youki, it has a leading `v`, whereas in runc it does not.(You can verify this by running the `state` command.)

https://github.com/youki-dev/youki/blob/main/crates/libcontainer/src/container/state.rs#L131

It looks like this value also seems to be `oci_spec::runtime::VERSION`.

Are you planning to examine the State-related parts around here later as well?

Reply:
Regarding state compatibility, I haven't completed the investigation yet. I'm planning to address this within the current PR, so I'll start working on it now.

For now, the version-related changes have been fixed in this commit: [7f4bf7d](https://github.com/youki-dev/youki/pull/3330/commits/7f4bf7da681100970781500dc198c8447fddb00a)

Reply:
Regarding the diff in the annotations (empty JSON during serialization), this seems to be an issue on the `oci-spec-rs` side. I'll address this separately.